src/Enum.php

```diff
@@ -175,13 +185,27 @@ public static function toArray()
      * @param $value
      * @psalm-param mixed $value
      * @psalm-pure
+     * @psalm-assert-if-true T $value
      * @return bool
      */
     public static function isValid($value)
     {
         return \in_array($value, static::toArray(), true);
     }
 
+    /**
+     * Asserts valid enum value
+     *
+     * @psalm-pure
+     * @psalm-assert T $value
+     */
+    public static function assertValidValue($value): void
+    {
+        if (!self::isValid($value)) {
```

Comment:
Since `isValid` are not private and not final - should we use `static::isValid` here?

Reply:
Changed